File: C++/binary_tree.cpp
#include <stdio.h>
#include <stdlib.h>
struct node
{
    int data;
    struct node *left, *right;
};
struct node *create()
{
    int x;
    struct node *newnode;
    newnode = (struct node *)malloc(sizeof(struct node));
    printf("Enter data (-1 for no node) : ");
    scanf("%d", &x);
    if (x == -1)
        return 0;
    newnode->data = x;
    printf("For left child of %d , ", x);
    newnode->left = create();
    printf("For right child of %d , ", x);
    newnode->right = create();
    return newnode;
}
void main()
{
    struct node *root;
    root = 0;
    root = create();
}

/* OUTPUT 
Enter data (-1 for no node) : 1
For left child of 1 , Enter data (-1 for no node) : 2
For left child of 2 , Enter data (-1 for no node) : 3
For left child of 3 , Enter data (-1 for no node) : -1
For right child of 3 , Enter data (-1 for no node) : -1
For right child of 2 , Enter data (-1 for no node) : 4
For left child of 4 , Enter data (-1 for no node) : -1
For right child of 4 , Enter data (-1 for no node) : -1
For right child of 1 , Enter data (-1 for no node) : 5
For left child of 5 , Enter data (-1 for no node) : -1
For right child of 5 , Enter data (-1 for no node) : -1
    Binary Tree :             1
                            /   \
                           2     5
                         /  \
                        3    4
*/

File: Python/binary_tree.py
class Node:
    def __init__(self, data):

        self.left = None
        self.right = None
        self.data = data

    def insert(self, data):
        if self.data:
            if data < self.data:
                if self.left is None:
                    self.left = Node(data)
                else:
                    self.left.insert(data)
            elif data > self.data:
                if self.right is None:
                    self.right = Node(data)
                else:
                    self.right.insert(data)
        else:
            self.data = data

    def display(self):
        if self.left:
            self.left.display()
        print(self.data)
        if self.right:
            self.right.display()


root = Node(0)
print("\n1. Insert an element\n2. Display the elements\n3. Exit\n")

while True:
    op = int(input("Enter your choice : "))
    if op == 1:
        root.insert(int(input("Enter your choice : ")))
    elif op == 2:
        root.display()
    elif op == 4:
        quit()
    else:
        print("Wrong choice !!! Try again")
